test/stache-test.js

```diff
@@ -2015,6 +2016,40 @@ function makeTest(name, doc, mutation) {
 
 		div = doc.createElement('div');
 
+		div.appendChild(renderer(data,{partials: partials}));
+		span = div.getElementsByTagName('span')[0];
+		strictEqual(domDataState.get.call(span, 'attr'), data.get('bar'), 'Nested data 1 should have correct data');
+
+		div = doc.createElement('div');
+		div.appendChild(renderer2(data,{partials: partials}));
+		span = div.getElementsByTagName('span')[0];
+		strictEqual(domDataState.get.call(span, 'attr'), data.get('bar'), 'Nested data 2 should have correct data');
+
+		div = doc.createElement('div');
+		div.appendChild(renderer3(data,{partials: partials}));
+		span = div.getElementsByTagName('span')[0];
+		strictEqual(domDataState.get.call(span, 'attr'), data.get('bar'), 'Nested data 3 should have correct data');
+	});
+
+	test("domData helper should set proper data instead of a context stack", function () {
+		var partials = {
+			'nested_data': stache('<span id="has_data" {{domData "attr"}}></span>'),
+			'nested_data2': stache('{{#this}}<span id="has_data" {{domData "attr"}}></span>{{/this}}'),
+			'nested_data3': stache('{{#../bar}}<span id="has_data" {{domData "attr"}}></span>{{/../bar}}')
+		};
+
+		var renderer = stache("{{#bar}}{{> nested_data}}{{/bar}}"),
+			renderer2 = stache("{{#bar}}{{> nested_data2}}{{/bar}}"),
+			renderer3 = stache("{{#bar}}{{> nested_data3}}{{/bar}}"),
+			div = doc.createElement('div'),
+			data = new SimpleMap({
+				foo: "bar",
+				bar: new SimpleMap({})
+			}),
+			span;
+
+		div = doc.createElement('div');
+
```

Comment:
This is a copy of the {{data}} tests, so in 5.x we can just drop the {{data}} tests in favor for these new {{domData}} tests.